zipline/assets/assets.py

```diff
@@ -544,6 +532,30 @@ def lookup_generic(self,
             self._lookup_generic_scalar(obj, as_of_date, matches, missing)
         return matches, missing
 
+    def map_identifier_list_to_sids(self, index, as_of_date):
```

Comment:
@ssanderson Does this method make sense? Its function is similar to lookup_generic, but also will build the assets that are needed.